scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,189 @@
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
+
+      // create a new tiled layer from the World Dark Gray Base REST service and set it as the scene's basemap
+      Basemap basemap = new Basemap(new ArcGISTiledLayer("https://services.arcgisonline" +
+        ".com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"));
+      scene.setBasemap(basemap);
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+      // store the newly created image overlay
+      imageOverlay = sceneView.getImageOverlays().get(0);
+
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files by file name in ascending order
+      if (imageFiles != null) {
+        imageFrames = Arrays.stream(imageFiles)
+          .sorted()
+          .map(f -> new ImageFrame(f.getAbsolutePath(), imageFrameEnvelope))
+          .collect(Collectors.toList());
+      }
+
+      startNewAnimationTimeline();
+
+    } catch (Exception e) {
+      // on any exception, print the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Set up a timeline to display the images at the specified speed from the combobox.
+   */
+  private void startNewAnimationTimeline() {
+
+    animation = new Timeline();
+    animation.setCycleCount(-1); // loop animation
+    animation.getKeyFrames().add(new KeyFrame(Duration.millis(period), e -> {
+      // set image frame to image overlay
+      imageOverlay.setImageFrame(imageFrames.get(frameIndex));
+      // update to the next frame
+      frameIndex = (frameIndex + 1) % imageFrames.size();
+    }));
+    animation.play();
+  }
+  
+  /**
+   * Controls the opacity of the image overlay using the slider.
+   */
+  @FXML
+  private void changeImageOverlayOpacity() {
+    imageOverlay.setOpacity((float) opacitySlider.getValue());
+  }
+
+  /**
+   * Handles the rate at which the image frames are displayed using the combo box.
+   */
+  @FXML
+  private void handleFramesComboBoxInteraction() {
+    // set the period for the chosen frame display speed 
+    switch (framesComboBox.getSelectionModel().getSelectedItem()) {
+      case "Fast":
+        period = 17;
+        break;
+      case "Medium":
+        period = 33;
+        break;
+      case "Slow":
+        period = 67;
+        break;
```

Comment:
we should really have a default case in this `switch`, you could just put it on 'slow'?

Reply:
I think because there's no other way any other value can be chosen in the combobox, we don't need a default here: it can only be one of "fast", "medium" or "slow".  In the code at the beginning we select index 2 from the combo box, and don't set that to anything else in the code (it would throw an error if we tried to set it to something beyond that index).